src/complex/Common/TypesUtility.hpp

```diff
@@ -374,6 +374,63 @@ inline constexpr DataType StringToDataType(std::string_view dataTypeString)
   }
 }
 
+/**
+ * @brief Returns a DataType for the passed in string representation
+ * @param dataTypeString
+ * @return
+ */
+inline constexpr DataType HumanStringToDataType(std::string_view dataTypeString)
```

Comment:
You could consider using an std::optional or nonstd::expected type rather than the exception? If we use the exception I think we are supposed to label the method as "throws" (Look up the spec or ping Jared for the specifics).